### DISCUSSION TITLE
A moral da pessoa desenvolvedora front-end nas equipes de engenharia está baixa

### DISCUSSION THREAD

**Discussion by woliveiras** [frontendbr/forum#1077] [Discussão]
Pessoal, em uma conversa no Telegram caímos na discussão sobre a "moral" da pessoa desenvolvedora front-end, onde uma grande parte das pessoas do grupo disse que passou por um mesmo problema: 

Em equipes de desenvolvimento com back-end, front-end e outros (UX, infra, mobile, etc.) a pessoa de back-end possui "mais moral" do que a de front-end em decisões que tocam diretamente o próprio front-end, como, por exemplo, arquitetura da aplicação front-end, qual framework usar, tooling, etc.

Inclusive foi relatado um caso recente onde a pessoa de front-end não foi ouvida em uma discussão técnica a respeito da divisão de sessões de um site.

O detalhe é que ninguém na conversa era iniciante/estagiário(a) na área.

Mais alguém já passou por esse tipo de situação, onde alguém de back-end "manda" mais na aplicação client-side do que a própria de front-end? - Se sim, compartilhem sua experiência.

**Comment by omariosouto** [comment]
Eu já cheguei a ouvir que o Front era a parte menos importante em um portal e que o Back tinha prioridade (usando um CMS, onde o trampo seria puxar os conteúdos cadastrados e não precisava modelar num nível absurdo o relacionamento das coisas no CMS, a estrutura tava praticamente pronta)

**Comment by klarkc** [comment]
Isso vem da cultura da empresa, que não valoriza UX, e por conseguinte não valoriza o trabalho da equipe de front end, que em diversas vezes pode ser tão ou mais importante que as outras equipes, principalmente para definir as regras de negócio. Na verdade o primeiro erro é dividir a empresa em "equipes front e back", isso em si já é completamente fora do desejável. Indo além, front-end e UX que delineiam as regras de negócios a partir dos desejos ou requisitos do usuário, não o 'CEO' da empresa ou a "equipe" de backend, afinal se o mercado determina como o produto deve ser, nada mais lógico que isso começar na experiência do usuário.

Minha sugestão é, se você não consegue mudar a cultura da empresa (trabalho bem difícil sem apoio), procurar um local onde o seu trabalho seja valorizado.

**Comment by mateusmiguel** [comment]
Ouvi do meu ex-patrão que o front era um mal necessário pra empresa.


Sério.

**Comment by woliveiras** [comment]
Não é meu caso hoje em dia, @klarkc. :joy: 

Foi uma conversa onde muitas pessoas disseram que passam por isso, inclusive eu já passei por esse tipo de situação em empregos anteriores.

**Comment by mahenrique94** [comment]
Aqui no meu trampo é bem tranquilo, tanto o front quanto back estão sempre conversando e trabalhando junto. Tudo que o pessoal do back vai fazer, eles me perguntam a melhor maneira e conversamos bastante, afinal, somos um time.

Acredito que ambos possuem o mesmo peso e devem andar junto, um não vive sem o outro, um back sem front, não será usado e um front sem back também (resumindo, ignorando postman ou html estáticos).

Mentalidades assim devem ser de pessoas antigas, onde, fazem aquelas telas e design ridiculos.

**Comment by eduardojmatos** [comment]
Isso rolava alguns anos atrás, hoje eu acho que isso diminuiu um pouco. E primeiro que, na minha opinião, algumas decisões precisam ser tomadas em conjunto (não todas, concordo, mas as que vão envolver mudanças de stack, alterações em API, etc., isso sim). Se tem alguém centralizando essas decisões no seu time, é hora de rever a forma que vocês decidem as coisas e como se consideram um time.

Eu confesso que a galera de front-end precisava começar a entender mais conceitos de desenvolvimento de software do que só ferramentaria. Infelizmente esse sentimento das outras pessoas de outras áreas ocorre principalmente por conta disso... Mas de novo, isso diminuiu bastante nos últimos anos, pelo menos na minha visão.

**Comment by eugeniobarrosjr** [comment]
Já tive uma discussão acalorada com meu P.O e o Líder do Back-End devido a melhorias que solicitei. Bateram o pé na decisão e pronto, mesmo sendo mudanças que iria melhorar e muito a experiência do usuário.

**Comment by woliveiras** [comment]
> Eu confesso que a galera de front-end precisava começar a entender mais conceitos de desenvolvimento de software do que só ferramentaria. 

Essa aqui toma +1 e dava até pra puxar de volta: https://github.com/frontendbr/forum/issues/568

**Comment by malaquiasdev** [comment]
Olá pessoas, invadindo o plantão de vocês aqui :)

Pra quem não me conhece, eu sou back-end \o/

Na minha opinião muita coisa que envolve engenharia DEVERIA ser decidida pelo TIME quando se há espaço na empresa para tal.

Mas aqui eu quero levantar uma questão, como o front-end vai defender o ponto de vista dele quando os times possuem mais back-ends do que front-ends? 

Por exemplo a ultima SQUAD que trabalhei: 1 front, 3 backs, 1 QA.

É uma dúvida minha, como 1 pessoa vai convencer 4? 

Nessas horas acredito que é preciso dominar conceitos e não só a pratica do que você ta fazendo. E acho que falta isso em alguns front-ends, dominam muito bem a prática e sabem a melhor maneira de aplica-la, isso não tem discussão...

Mas pecam na hora de provar seu ponto de vista por não saber como se expressar na maioria das vezes.

Eu não quero levantar **treta** aqui, mas diferente do que o @eduardojmatos levantou, hoje em dia estou vendo mais pessoas desenvolvedoras focadas na prática e na melhoria constante do seu ferramental e não sabendo se expressar ou defender seu ponto de vista.

Pro caso levantado pelo @woliveiras talvez uma TRIBO de front-ends resolvesse o caso, lá na empresa a arquitetura de front-end foi definida no começo por back-ends e estava completamente cadaga, com a contratação de novos front-ends, eles ganharam números, eles resolveram entre eles criar uma TRIBO pra discutir essas coisas e começaram a implementar.

Eu acho que deu muito certo! E back passou a interagir só nó que diz respeito a integração entre as camadas.

**Comment by woliveiras** [comment]
> Pro caso levantado pelo @woliveiras talvez uma TRIBO de front-ends resolvesse o caso, lá na empresa a arquitetura de front-end foi definida no começo por back-ends e estava completamente cadaga, com a contratação de novos front-ends, eles ganharam números, eles resolveram entre eles criar uma TRIBO pra discutir essas coisas e começaram a implementar.

Muito bom compartilhar essa experiência, mano. Valeu!

**Comment by diessica** [comment]
>  eles resolveram entre eles criar uma TRIBO pra discutir essas coisas e começaram a implementar.

um "chapter" então!
> Chapters are a group or team members working within a special area. So for example a squad might be made up of front office developers, back office developers, database administrator and testers. So a chapter could be a ‘front office chapter’, where front office developers get together and exchange ideas, get help on challenges and discuss new technologies. For example one developer might start using AngularJS on a new feature, and will relay to the rest of the chapter his experience and discuss on how other squads can use it. This is a great way to promote innovation and ‘cross pollination’ of ideas across teams. 

- http://www.full-stackagile.com/2016/02/14/team-organisation-squads-chapters-tribes-and-guilds/

**Comment by caioincau** [comment]
Tem a questão cultural sim, até pq tem mto back-end que acredita ser fullstack, faz umas gambiarras com jQuery e na visão dele, o front só tá lá para fazer um trampo que para ele “é fácil”, mas é chato.

Mas também tem o caso que o Edu citou, tem muito front aí que não tem uma noção mínima de arquitetura de aplicação, seja ela rest ou não, não sabe nada de banco de dados, aí quando rola uma discussão técnica, não consegue explicar bem as mudanças necessárias e o impacto delas, pior, às vezes falando errado(o que deveria ser normal, todos erramos), mas em meio a uma discussão, isso pode acabar sendo mal interpretado como um cara que não sabe o que está falando.

**Comment by woliveiras** [comment]
Para contextualizar melhor: as discussões são sobre client-side mesmo, não sobre uma aplicação inteira (front + back).

Imagina uma discussão entre usar Grunt ou NPM scritps, React ou jQuery, usar uma tag <button> ou <a>, mas "quem manda" é a pessoa de back-end e a de front não é ouvida, mesmo que a de front-end deveria ser a mais respeitada em seu domínio de atuação.

Acredito que pelo fato da pessoa de front não se expressar pode acontecer, sim, de não ser ouvida. Mas o que temos hoje no mercado é uma visão de que o profissional que realmente sabe software/sabe programação é o backendero, não o frontendero, independente da discussão que esteja rolando.

Temos professores de faculdade que não dominam o mínimo de frontend e chamam front de sub-emprego, trazendo a essa mentalidade para a maioria dos profissionais.

O mesmo acontece na liderança de equipes, onde, em sua maioria, são backenderos e não frontenderos, fazendo com que o frontend seja desconsiderado também.

Tirando casos raros, não podemos tratar exeção como regra, esse é o cenário que temos no mercado de trabalho. Quando saímos de São Paulo então vemos muito mais disso.

**Comment by wilkerfoureaux** [comment]
tem que levar em conta o numero de empresas que ainda dependem diretamente dos devs e a cultura geral de que não se ligava muito mesmo para o front-end... QA e front-end sempre eram tratados como "acabamento", sem contar a deadline...

tudo está ligado à maturidade da empresa e budget do cliente... na minhã visão... ainda vai se ouvir muito relatos como deram acima. Quando digo maturidade, penso na questão de evitar problemas antecipadamente com base no skill de TODO o time, seja junior ou senior, tudo conta.

acho que nem é pra desanimar não... se tiverem forças pra fazer um workshop, uma talk, um artigo, algo que demonstre os benefícios, já é um passo a ser dado, se já não foi feito. Senão é partir pra outra mesmo...

parece que não, mas tem um gap enorme de comunicação sempre, quando o assunto é front, priorização errada, visão simplista (tipo, "ah, é só copiar aquele componente e pronto", coisas do tipo que fazem esmorecer...

tem que saber "vender o peixe", sempre...

**Comment by vinicius73** [comment]
Talvez eu esteja apenas chovendo no molhado, mas o @caioincau tocou em algo **critico**, assim como o @eduardojmatos 

Há muitas pessoas _front-ender_ que só querem saber como _usar_ o seu ferramental novo.
Enquanto _back-enders_, menosprezam tudo o que o front-end se tornou.

Muitas vezes a arquitetura fron-end é tão complexa quanto a do back-end, eu diria até mais complexas em alguns aspectos... (assunto para outro momento)

Ambos os lados devem aprender a importância de cada camada. Um não é capaz de viver sem o outro.

- Do que adianta um software que _funciona_ mas não é _funcional_ para o cliente?
- Do que adianta um software _bonito_ que não _resolve_ o problema?
- Do que adianta um banco de dados todo normalizado que não da as informações que o cliente quer ver?
- Do que adianta uma api _documentada_ que sobrecarrega do _client_
- Do que adianta entregar o novo end-point se é necessário disponibilizar a feature em 3 versões diferentes do software para 3 plataformas diferentes? 

São coisas do dia-a-dia que não podem existir sem ambas as partes. 
Nada de _utopias_ fronendless ou backendless.

A raiz do problema esta no *comportamento humano*, na falta de empatia. 
Reparem que é a mesma discussão Fron-end VS Designer, isso só _parou_ quando um entendeu como o trabalho do outro funciona.

Para mim, não importa qual o _chapéu_ a pessoa vai vestir, se estamos falando de desenvolvimento de software, todos devem focar em entregar o melhor da melhor forma para tanto *equipe* quanto *cliente* fiquem felizes.

Então, não adianta frond-end reclamar que back-end é complicado... ou back-end reclamar que front-end é firula... Se qualquer dessas partes tiver problemas o cliente tem problemas, e com isso a empresa tem problemas.  
Não sei vocês, mas pra mim a conta é simples: se a empresa tiver problemas, eu terei problemas em breve, pois ela não vai poder pagar meu salário :wink:

**Comment by bernardodiasc** [comment]
Front-end pra que? 👉 [getbootstrap.com](https://getbootstrap.com/) 😂 

Brincadeiras a parte, muito boa essa discussão. To seguindo.

Minha humilde opinião é que "profissionais" que menosprezam o trabalho alheio são bobos. 😄

**Comment by davidallysson** [comment]
> Temos professores de faculdade que não dominam o mínimo de frontend e chamam front de sub-emprego, trazendo a essa mentalidade para a maioria dos profissionais.

E o que vocês acham a respeito da falta de uma graduação focada em front-end? Digo, aqui na minha cidade (Natal, RN) só temos graduações focadas em Desenvolvimento num geral (onde se vê back em sua maioria e um pouco do front).

Muitos amigos meus que são fronts sentem falta de uma graduação direcionada para as necessidades do mercado de front. Algo mais específico e mais direcionado.

A falta de um curso assim não alimenta ainda mais essa cultura não? @woliveiras

**Comment by didiraja** [comment]
@davidallysson nunca tinha pensando nisso, mas faz sentido. Pensando em graduação, são sempre cursos de programação totalmente back-end, as vezes até de maneira antiquada ou no máximo, alguns cursos de Design mais amplos que passam cosia de web e que, mesmo assim, é uma pincelada bem curta (como foi o meu).

Um curso que fosse metade fundamentos do design e metade html/css/js em si, seria um ótimo curso formador de front-enders e até ui/ux.

**Comment by woliveiras** [comment]
@davidallysson, não acredito que as faculdades sejam focadas em backend. Elas ensinam o que precisamos saber de ciência da computação, analise de sistemas, engenharia da computação, etc. sabe?

Bancos de dados, server side, redes... Isso é como sistemas funcionam, não é só backend. Acredito que todo mundo precisa saber isso, como a galera comentou acima. A não ser que o objetivo da pessoa não seja trabalhar com software, mas com design, por exemplo.

Eu faço análise e desenvolvimento de sistemas e, até agora, não tive backend, tive matérias relacionadas a sistemas: lógica, s.o., redes, estruturas de dados, bancos de dados, engenharia de software, OOP.

Mas não tivemos nenhuma matéria que focasse em criar sistemas server-side. Tivemos Java, mas o objetivo era aprender OOP com ela, também tivemos C, mas o objetivo era aprender algoritmos... Assim vai.

**Comment by eduardojmatos** [comment]
> Eu não quero levantar treta aqui, mas diferente do que o @eduardojmatos levantou, hoje em dia estou vendo mais pessoas desenvolvedoras focadas na prática e na melhoria constante do seu ferramental e não sabendo se expressar ou defender seu ponto de vista.

@mmalaquiasdev mas vc não falou nada diferente do que citei huahuahua. Galera muito focada em ferramentaria, isso que quis dizer :)

> E o que vocês acham a respeito da falta de uma graduação focada em front-end? Digo, aqui na minha cidade (Natal, RN) só temos graduações focadas em Desenvolvimento num geral (onde se vê back em sua maioria e um pouco do front).

@davidallysson Eu acho too much. Pra mim, ainda faz sentido galera de front-end fazer uma Ciência da Computação, ou Análise de Sistemas, ou qualquer graduação do tipo pra ter essa base. Vejam que não estou falando que é preciso disso pra ser uma pessoa desenvolvedora front-end, mas ajuda bastante estudar conceitos de desenvolvimento de software fora da caixinha do client-side.

**Comment by vinicius73** [comment]
__Voto com o relator__ @eduardojmatos
Front-end é software.
Não deve ser tratado de outra maneira, o resto são tecnologias, técnicas, APIs... Coisa que faculdade nenhuma vai (e nem tem como) ensinar


Vale ainda dizer que front-end não é sinônimo de html, css e JavaScript...

**Comment by torto** [comment]
Acredito que não necessite fazer qualquer universidade para saber programar, mas gostaria de deixar minha humilde opinião: Falta base teórica para muitos dos profissionais que trabalhei (e me incluo nessa). E todos sabemos que teoria não é bobeira e te torna um programador melhor :heart: 

E mano, trampei em uma empresa muito boa em JS e te garanto... a galera tem usado muito dos padrões de projeto que usamos em Java (we are the world :earth_americas:)

Sem contar que hoje, muito da parte da lógica de negócio esta no front.. ou seja... empresas que não vê valor nisso precisam urgentemente se atualizarem pq se não serão o novo Delphi!

Dica free: Caso a empresa seja escrota: Procure um outro emprego que vc vai ser mais feliz!

**Comment by klarkc** [comment]
TLDR: Concordo que falta base teórica, principalmente de engenharia, e pela maioria das vezes isso não pode ser visto na academia. Mas essa falta, não ocorre apenas no profissional de front-end, ocorre em programadores de maneira geral - é um ciclo natural da carreira de qualquer pessoa que trabalha na área de software.

Olhando pra mim mesmo, acho que todos que têm dificuldade em justificar arquitetura, funcionalidades, configurações e etc, o fazem porque falta conhecimento da engenharia. E ela é sim necessária para o profissional que quer passar da parte de execução para a parte de planejamento. Só é possível justificar o uso de algo quando você sabe o porquê está usando este algo. É o mesmo caso do engenheiro e do mestre de obras.

Quanto a engenharia uma boa parcela aprendemos sim na academia, pois conceitos ali abordados foram atualizados e são usados como fundamentos e são pilares de muita coisa hoje, não sei se é a melhor forma de aprender, mas ela acrescenta muito conhecimento ao profissional - não dá pra negar isso. Porém também, apenas a academia não ensina outra grande fatia dessa pizza. A realidade e o mercado impulsionam a engenharia de software de formas que a academia nunca vai poder estudar de forma efetiva, pois até mesmo a velocidade em que isso ocorre torna inviável o seu estudo. Entra aí um pouco de feeling, experiência, métricas, conhecimento prático e teste/validação em várias tomadas de decisão.

**Comment by akfzambrana** [comment]
Esse "desprezo" que o @woliveiras fala eu sinto na pele inumeras vezes. Trabalhei em uma empresa que o pessoal de front precisava de aprovação do PR de uma pessoa core-commiter, mas não tinha core-commiter frontender, apenas o pessoal do backend era core-commiter. Isso era extremamente frustrante.
Sobre a questão de maior conhecimento de engenharia de software, eu acho que todas as pessoas que trabalham com desenvolvimento precisam primeiro de:
- *respeito* pelo trabalho das outras pessoas o que resolve de certa forma esse desprezo
- *conceitos de engenharia de software* sem importar ferramental.

E sim, na maioria das empresas que conheço/trabalhei, lider tecnico de front é backend e quem escolhe as tecnologias do front tb é backend ou uma pessoa arquiteta que nem sabe como funciona o "client side" das aplicações.

**Comment by synga** [comment]
Cara isso é chato demais e envolve muita coisa além de cultura, apesar de eu achar que cultura é predominante e a mudança disso caminha a passos curtos, talvez a galera de SP sinta mais, mas pro interior aqui na maioria o backend "manda" em tudo e é sempre a voz da razão.
Já passei por isso quando era estag/trainee, os caras falavam que teria um projeto novo e só me incluíam nas reuniões quando já estava TUDO definido, não me perguntavam se meu conhecimento com ferramenta X ou Y estava bom o suficiente pra começar o projeto nem nada. Acabava atrasando o projeto por que tinha que gastar um tempo pra me familiarizar com aquilo e a cobrança era enorme.

O backend escolher a stack que você vai trabalhar é realmente chato, mas na maioria das vezes já vem na descrição da vaga que você PRECISA saber aquilo, vai ter que trabalhar com aquilo, agora se colocarem, por exemplo, **"conhecimento em frameworks (Angular, Vue, React, etc.)"** ai já é outra coisa. Agora se o cara vem com **"Não quero SASS, quero metodologia tal, quero linter tal, quero task manager tal, teste só se for com X e Y"** ai já é caso de ficar fulo da vida.

Isso do dev de front ser mais focado em ferramentas e não em arquitetura/engenharia e deveriam melhorar nisso pra poder serem mais ouvidos não sei se cola, até por que a maioria dos backs entendem quase nada de front e podem dar pitaco, se um não vale o inverso disso também não, certo? Eles teriam que aprender sobre frontend/UX pra poder falar algo? Cada um tem que ter voz no que tem expertise, afinal é um time.

Maaaaaas claro que ter esse conhecimento de arquitetura/engenharia é sempre bem vindo, ao menos para fazer as pessoas questionarem se o que você falou faz sentido. Você não precisa entender o A+B profundamente técnico do por que o cara do backend está viajando nas ideias, mas se você consegue fazer ele e o time se questionarem já está ótimo.

A galera não está acostumada a bater o pé pras ideias que tem. Teve aquela thread (#911) falando sobre o que falta nos candidatos que as empresas entrevistavam e uma das coisas mais mencionadas foi a falta de saber se expressar. A galera engenheira exercita isso com muito mais frequência, enquanto o dev de front esta ali, na maioria das vezes, só de olho no que ele vai ter que fazer e levanta uma questão ou outra. Se você não levantar da cadeira pra falar **"vai dar merda isso ai"** e saber expor sua opinião ninguém vai te ouvir. Tem que ter argumentos, tentar provar e falar o por que aquilo te afeta, como te afeta e como poderia ser melhor. Se os caras não aceitarem, desrespeitarem sua opinião e você ver que caiu e novo nessa coisa de "backend manda" então meu, senta, fica de boa, faz e espera dar ruim. Você fez sua parte, certo? Tentou o melhor pro projeto não foi?

No demais, como já falaram ai, não ta feliz? Não se sente valorizado? Não é ouvido? Não exercita essas coisas? Vai atrás de outro lugar pra trabalhar. E cabe a nós acabarmos com essa cultura, certo? Temos que fazer diferente quando isso acontecer para que um dia não aconteça mais.

**Comment by felipefialho** [comment]
Demorei uns dias para participar, porque estava lendo com calma os comentários de vocês.

Esse era meu sentimento anos atrás, Desenvolvimento Front-end de fato era tratado como simples e em muitos casos deixavam alguém cuidando disso para que __Back-enders__  não precisassem parar suas tasks e mexer com HTML e CSS (e muitas vezes eles não curtiam).

Os anos foram passando, a profissão de  __Front-end Developer__ foi crescendo e ganhou o mesmo grau de importância do que outras áreas de desenvolvimento, isso porque, se tratando de desenvolvimento web, TODAS tem sim o mesmo nível de importância dentro de um projeto. E isso inclui UX, Design e todas as pessoas que participam do processo de desenvolvimento de um produto.

Infelizmente nem todas as empresas tem esse grau de maturidade, e ainda acontecem situações como as descritas pela @akfzambrana e pelo @mateusmiguel, mas isso diminuiu nos últimos anos e a tendencia é esse comportamento sumir.

Um ponto que quero levantar, é que mesmo profissionais __Front-enders__ costumam agir com extrema marra e desdém quando mudam de área ou mesmo se tornam full-stacks. 

Devem lembrar de várias pessoas que se destacaram como __Front-enders__, construiram toda uma carreira nessa profissão... e suas primeiras atitudes quando migraram para outras áreas ou ganharam outras skills como developers, foi tecer comentários diminuindo o desenvolvimento Front-end.

**Comment by bernardodiasc** [comment]
Acho legal referenciar essa https://github.com/frontendbr/forum/issues/707, pois o contexto histórico pode ser relevante pra discussão.

**Comment by caioincau** [comment]
"Devem lembrar de várias pessoas que se destacaram como Front-enders, construiram toda uma carreira, e suas primeiras atitudes quando migraram para outras áreas ou ganharam mais skills como developers, foi tecer comentários diminuindo essa profissão." - @LFeh 

Vários, infelizmente, cospem no prato que comeram.

**Comment by woliveiras** [comment]
> Devem lembrar de várias pessoas que se destacaram como Front-enders, construiram toda uma carreira, e suas primeiras atitudes quando migraram para outras áreas ou ganharam mais skills como developers, foi tecer comentários diminuindo essa profissão.

Excelente observação, mano @LFeh!
